opstack/src/server/mod.rs

```diff
@@ -62,18 +85,44 @@ async fn chain_id_handler(State(state): State<Arc<RwLock<ServerState>>>) -> Json
     Json(state.read().await.chain_id)
 }
 
+async fn unsafe_signer_proof_handler(
+    State(state): State<Arc<RwLock<ServerState>>>,
+    Path(block_hash): Path<FixedBytes<32>>,
+) -> Result<Json<EIP1186AccountProofResponse>, StatusCode> {
+    let rpc = HttpRpc::<Ethereum>::new(&state.read().await.execution_rpc.to_string())
+        .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
+    let signer_slot =
+        B256::from_str(UNSAFE_SIGNER_SLOT).map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
+    let block = rpc
+        .get_block(block_hash)
+        .await
+        .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)?;
+    let proof = rpc
+        .get_proof(
+            state.read().await.system_config_contract,
+            &[signer_slot],
+            block.number.to(),
```

Comment:
Also I think its possible to pass a blockhash in here. We should prefer that in case there is a fork and multiple blocks exist at a given block number.

Reply:
Makes sense!